ompi/mca/coll/base/coll_base_topo.c

```diff
@@ -59,7 +59,12 @@ static int calculate_num_nodes_up_to_level( int fanout, int level )
 {
     /* just use geometric progression formula for sum:
        a^0+a^1+...a^(n-1) = (a^n-1)/(a-1) */
-    return ((pown(fanout,level) - 1)/(fanout - 1));
+    if (1 == fanout) {
+        return pown(fanout,level);
+    }
```

Comment:
This is 1 ^ level, so it is safe to return 1.

Reply:
Missed that, sorry, fixed now.

Reply:
Correct, `level` is the number of predecessors in the chain.